### PR TITLE
Try out constant_keyword value fields in system log integrations

### PR DESCRIPTION
**Pull request by fearful-symmetry**
## What does this PR do?

This is a draft PR to test out the changes discussed here: https://github.com/elastic/security-team/issues/780
This is an addition based on the spec changes here: https://github.com/elastic/package-spec/pull/194

This probably won't pass CI or anything now, as we're also waiting for this: https://github.com/elastic/elastic-package/pull/386

This PR adds the `value` field to `constant_keyword` fields in the base yaml files.

## Checklist

- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] If I'm introducing a new feature, I have modified the Kibana version constraint in my package's `manifest.yml` file to point to the latest Elastic stack release (e.g. `^7.13.0`).

### CODE REVIEW

**Comment by peluja1012** [comment]
Hi @fearful-symmetry, thanks for opening up this PR! From the Security Solution side, the most important piece is for this integration to populate both `event.dataset` and `event.module`. The idea, as described [here](https://github.com/elastic/security-team/issues/780#issuecomment-858685533), was that integrations, such as this one, could populate values for these fields by defining them as `constant_keyword` and then setting the value in the index mapping. I'm not too familiar with this codebase, but in this PR I don't see any changes related to `event.dataset` or `event.module`. If values for these fields are already being set somewhere else, then we're probably ok. But otherwise, we may need to make other updates here or somewhere else.

**Comment by andrewkroh** [comment]
Thanks for setting up a test. I merged support for the `value` field into Kibana so after we get some new snapshots I expect this should work. https://github.com/elastic/kibana/pull/103000

**Comment by fearful-symmetry** [comment]
Figured I'd get the fields mixed up. Let's try that.

**Comment by fearful-symmetry** [comment]
@andrewkroh do we want to manually set the value for `event.dataset` or use some kind of alias?

File: packages/system/data_stream/auth/fields/base-fields.yml [hunk 1]
```diff
@@ -1,12 +1,22 @@
 - name: data_stream.type
   type: constant_keyword
   description: Data stream type.
+  value: logs
```
Comment (ruflin) [inline packages/system/data_stream/auth/fields/base-fields.yml:4]: This is a nice addition we likely should add anywhere. @mtojek I remember we had some discussion around this in the past, maybe there is even an issue for it. Later on we should make sure it is in sync with the type in the manifest.

File: packages/system/data_stream/auth/fields/base-fields.yml [hunk 2]
```diff
@@ -1,12 +1,22 @@
...
 - name: data_stream.dataset
   type: constant_keyword
   description: Data stream dataset.
+  value: system.auth
+- name: event.dataset
```
Comment (ruflin) [inline packages/system/data_stream/auth/fields/base-fields.yml:9]: Nit: Can we move event.dataset and event.module to the bottom to have it together as one block? Like this the diff is also cleaner to only have an addition on the bottom, assuming we leave out the changes for value in data_stream.*

**Review by mtojek — COMMENTED** [review]
Please update the changelog and version in the package manifest.

File: packages/system/data_stream/auth/fields/base-fields.yml (hunk 1 above)
Comment (mtojek) [reply packages/system/data_stream/auth/fields/base-fields.yml:4]: Are you referring to validation only? If so, then I understand that this property will ALWAYS be in sync with the type (any overriding forbidden)?

Comment (ruflin) [reply packages/system/data_stream/auth/fields/base-fields.yml:4]: Yes, can't think of a use case where the two would not be in sync.

**Comment by ruflin** [comment]
For reference, here the two related issues: https://github.com/elastic/integrations/issues/226 https://github.com/elastic/kibana/issues/66551

File: packages/system/data_stream/security/fields/base-fields.yml [hunk 3]
```diff
@@ -1,21 +1,33 @@
...
   type: constant_keyword
   description: Data stream type.
+  value: logs
 - name: data_stream.dataset
   type: constant_keyword
   description: Data stream dataset name.
+  value: system.security
+- name: event.dataset
+  type: constant_keyword
+  description: event dataset.
```
Comment (mtojek) [inline packages/system/data_stream/security/fields/base-fields.yml:11]: nit: should be uppercase

File: packages/system/data_stream/security/fields/base-fields.yml [hunk 4]
```diff
@@ -1,21 +1,33 @@
...
+  value: system.security
 - name: data_stream.namespace
   type: constant_keyword
   description: Data stream namespace.
 - name: dataset.type
```
Comment (mtojek) [inline packages/system/data_stream/security/fields/base-fields.yml:16]: @ruflin shouldn't this field be removed?

File: packages/system/data_stream/security/fields/base-fields.yml [hunk 5]
```diff
@@ -1,21 +1,33 @@
...
   type: constant_keyword
   description: Dataset type.
+  value: logs
 - name: dataset.name
```
Comment (mtojek) [inline packages/system/data_stream/security/fields/base-fields.yml:20]: @ruflin shouldn't this field be removed? I think we replaced it with data_stream

File: packages/system/data_stream/security/fields/base-fields.yml [hunk 6]
```diff
@@ -1,21 +1,33 @@
...
   type: constant_keyword
   description: Dataset name.
+  value: system.security
 - name: dataset.namespace
```
Comment (mtojek) [inline packages/system/data_stream/security/fields/base-fields.yml:24]: @ruflin same here

**Comment by fearful-symmetry** [comment]
Alright, everything should be updated now. Just depends on what we want to do with the `value` in the `data_stream` objects that I added because I was confused, as well as `data_stream.type`

**Comment by fearful-symmetry** [comment]
A tad confused by the CI failures, since `ep check -v` seems happy when I run it locally.

**Comment by fearful-symmetry** [comment]
...helps if I update `elastic-package`

File: packages/system/changelog.yml [hunk 7]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "0.13.5"
+  changes:
+    - description: Use event.datastream and event.module
```
Comment (peluja1012) [inline packages/system/changelog.yml:4]: Should this say `event.dataset` instead of `event.datastream`?

Comment (fearful-symmetry) [reply packages/system/changelog.yml:4]: Good catch!

**Review by mtojek — COMMENTED** [review]
Could you please make sure that `dataset.*` are not required anymore and clean them? I saw some in https://github.com/elastic/integrations/blob/27d37a116d765c32cccbbfd4f0773e563aed093c/packages/system/data_stream/security/fields/base-fields.yml .

**Comment by fearful-symmetry** [comment]
> Could you please make sure that dataset.* are not required anymore and clean them? 

@mtojek that's one of the things I'm not clear on. What uses `dataset.*`? I'm not really sure if it's safe to remove.

**Comment by fearful-symmetry** [comment]
Okay, so it just now occurred to me that when you said "required anymore" you were probably referring to the dashboards & saved searches. If that's the case, no, nothing is using them.

**Review by mtojek — APPROVED** [review]
LGTM

**Pull request merged**